src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Resources/MarkAssembliesWithNeutralResourcesLanguage.cs

```diff
@@ -53,9 +53,26 @@ public override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationStartAction(cc =>
             {
-                var hasResource = false;
+                INamedTypeSymbol? generatedCode = cc.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemCodeDomCompilerGeneratedCodeAttribute);
```

Comment:
[cosmetic][minor] You could use the `TryGetXXX` instead of doing get then null check.